app/src/org/commcare/network/HttpRequestGenerator.java

```diff
@@ -68,15 +68,25 @@
      */
     public static final String AUTH_REQUEST_TYPE_NO_AUTH = "noauth";
 
+    private static final String SUBMIT_MODE = "submit_mode";
+
+    private static final String SUBMIT_MODE_DEMO = "demo";
+
     private Credentials credentials;
     PasswordAuthentication passwordAuthentication;
     private String username;
+    private String userType;

```

Comment:
Would argue that it would be cleaner to replace this with `private final boolean isDemoUserRequest`.
(requires you to set the field in the empty constructor, which I think is fine)